src/bigquery.ts

```diff
@@ -2217,15 +2217,13 @@ export class BigQueryTimestamp {
       if (/^\d{4}-\d{1,2}-\d{1,2}/.test(value)) {
         pd = new PreciseDate(value);
       } else {
-        const floatValue = Number.parseFloat(value);
-        if (!Number.isNaN(floatValue)) {
-          pd = this.fromFloatValue_(floatValue);
-        } else {
-          pd = new PreciseDate(value);
-        }
+        pd = new PreciseDate(BigInt(value) * BigInt(1000));
       }
+    } else if (value > 0) {
+      pd = new PreciseDate(BigInt(value) * BigInt(1000));
     } else {
-      pd = this.fromFloatValue_(value);
+      // Nan or negative values - invalid dates
```

Comment:
negative values are still valid, they just mean they're prior to the unix epoch.  Consider this:
`SELECT CAST('1969-12-25 00:00:00' as TIMESTAMP) as ts`

Reply:
Added tests for negative unix timestamps.